### PR TITLE
Finish create feedback service layer 

### PR DESCRIPTION
**Pull request by shaharshw**
## Description

It adds FeedbackCreatorService, FeedbackFetcherService, and FeedbackUpdateService and its tests.

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackCreatorService.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.hibob.academy.employeeFeedback.service.feedback
+
+import com.hibob.academy.employeeFeedback.dao.FeedbackDao
+import com.hibob.academy.employeeFeedback.model.CreateFeedbackRequest
+import com.hibob.academy.employeeFeedback.utils.getLoggedInUser
+import org.springframework.stereotype.Service
+
+@Service
+class FeedbackCreatorService(
```
Comment (rakreshet) [inline src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackCreatorService.kt]: let's rename, not need the word `Service` at all

File: src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackCreatorService.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    private val feedbackDao: FeedbackDao
+) {
+
+    fun createFeedback(createFeedbackRequest: CreateFeedbackRequest) : Long {
```
Comment (rakreshet) [inline src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackCreatorService.kt]: I expect that the loggedIn user will be passed as a parameter from the controller

File: src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackFetcher.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+@Service
+class FeedbackFetcher(
+    private val feedbackDao: FeedbackDao
+) {
+
+    fun getFeedbackById(loggedInUser: LoggedInUser, feedbackId: Long) : Feedback {
+        return feedbackDao.getFeedbackById(loggedInUser.companyId, feedbackId)
+    }
+
+    fun getFeedbacks(loggedInUser: LoggedInUser, filters: FilterFeedbackRequest) : List<Feedback> {
```
Comment (rakreshet) [inline src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackFetcher.kt]: don't we want to be able to fetch also by userId?

Comment (shaharshw) [reply src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackFetcher.kt]: maybe, but it wasn't in the specs

File: src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackFetcher.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    fun getFeedbackById(companyId: Long, feedbackId: Long) : Feedback {
+        return feedbackDao.getFeedbackById(companyId, feedbackId)
+    }
+
+    fun getFeedbacksByCompany(companyId: Long, filters: FilterFeedbackRequest) : List<Feedback> {
+        return feedbackDao.getFeedbacksByFilters(companyId, filters)
+    }
+
+    fun getFeedbackStatusById(companyId: Long, feedbackId: Long) : StatusResponse {
```
Comment (rakreshet) [inline src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackFetcher.kt:23]: there's a little redundancy. It's ok just to use `getFeedbackById` and the controller will just return the status. 
I don't remember the dao method - do you use a different mapper just to extract the status?
Anyhow - I think you can just remove this method. Up to you

**Review by rakreshet — APPROVED** [review]

File: src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackFetcher.kt (hunk 4 above)
Comment (shaharshw) [reply src/main/kotlin/com/hibob/academy/employeeFeedback/service/feedback/FeedbackFetcher.kt:23]: I have a different mapper for the status and a different method for this in the dao.

**Pull request merged**